src/main/resources/static/js/submitFormBtn.js

```diff
@@ -0,0 +1,33 @@
+$(document).ready(function () {
+    $('#submitFormBtn').click(function () {
+        var formData = $("#subjectForm").serialize();
+
+        $.ajax({
+            type: "POST",
+            url: "/userSubject/save",
+            data: formData,
+            success: function (response) {
+                if (response.success) {
+                    window.location.href = "/userSubject/selected";
+                } else {
+                    var errorMessages = (response.errors || []).join("<br>");
+                    $('#errorMessage').html(errorMessages);
+                    $('#errorModal').modal('show');
+                }
```

Comment:
as I explaind previously, there is never the case of success == true in the success-function.
refactor to only use the response-status (success-cases eg 200 and error-casses eg 400) to determine if the submit was successfull. thereby you can also remove the unnecessary parts in userSubject controller

Reply:
I've modified and pushed, I think its old code